File: other/twoSum.cpp
// leetcode 1. two sum

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        vector<int> res;
        map<int, int> hashmap;
        for (int i = 0; i < nums.size(); i++)
        {
            hashmap.insert(map<int, int>::value_type(nums[i], i));
        }
        for (int i = 0; i < nums.size(); i++)
        {
            int diff = target - nums[i];
            if (hashmap.count(diff) > 0 && i != hashmap[diff])
            {
                res.emplace_back(i);
                res.emplace_back(hashmap[diff]);
                return res;
            }
        }
        return res;
    }
};

File: other/twoSum.py
# leetcode 1. two sum

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        for i, num in enumerate(nums):
            dic[num] = i
        for i, num in enumerate(nums):
            diff = target - num
            if diff in dic and dic[diff] != i:
                return [i, dic[diff]]
